remhos_lo.cpp

```diff
@@ -942,4 +944,456 @@ void PAResidualDistribution::CalcLOSolution(const Vector &u, Vector &du) const
    });
 }
 
+//====
+//PA Residual Distribution Subcell
+//
+PAResidualDistributionSubcell::PAResidualDistributionSubcell
+(ParFiniteElementSpace &space,
+ ParBilinearForm &Kbf,
+ Assembly &asmbly,
+ const Vector &Mlump,
+ bool subcell, bool timedep)
+   :  PAResidualDistribution(space, Kbf, asmbly, Mlump, subcell,
+                             timedep), init_weights(true)
+{
+}
+
+void PAResidualDistributionSubcell::SampleSubCellVelocity() const
+{
+   const int dim = assembly.GetSubCellMesh()->Dimension();
+   const IntegrationRule *ir;
+   if (dim == 2) { ir = &IntRules.Get(Geometry::SQUARE, 1); }
+   if (dim == 3) { ir = &IntRules.Get(Geometry::CUBE, 1); }
+
+   Mesh *mesh = assembly.GetSubCellMesh();
+   const int NE = mesh->GetNE();
+   const int nq = ir->GetNPoints();
+
+   //Q: which to use 0, or 1?
+   FiniteElementSpace *SubFes = assembly.lom.SubFes0;
```

Comment:
@vladotomov , Here I am not too sure which space I should use 0 or 1? Perhaps it doesn't matter?

Reply:
Doesn't matter, the code below this uses only the mesh of `SubFes`, it doesn't use any space info.